### ISSUE TITLE
curl seems to ignore netrc in version 7.63

### ISSUE THREAD

**Issue by justinvreeland** [curl/curl#3455]
### I did this
Ran 'git fetch --all' on a machine that uses .netrc and kinit 

Netrc contents:

~~~
 default
  login <username>
  password _
~~~

### I expected the following
git to fetch the new contents as it does when curl 7.61 and 7.62 and previous are installed.

instead i'm prompted for a username.

### curl/libcurl version
 curl -V
`curl 7.63.0 (x86_64-pc-linux-gnu) libcurl/7.63.0 OpenSSL/1.1.1a zlib/1.2.11 libidn2/2.0.5 libpsl/0.20.2 (+libidn2/2.0.4) libssh2/1.8.0 nghttp2/1.35.1
Release-Date: 2018-12-12
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL 
`

### operating system
ArchLiux

**Comment by bagder** [comment]
I fail to reproduce. I did:
~~~
$ cat test_netrc
default
 login user
 password yes
$ curl --netrc-file test_netrc localhost -v 2>&1 | grep Basic
* Server auth using Basic with user 'user'
> Authorization: Basic dXNlcjp5ZXM=
~~~

**Comment by justinvreeland** [comment]
I reran the commands with GIT_CURL_VERBOSE=1.

The only differences between them (aside from dates and Authorization/Authenticate values) is that the working version of curl (7.61 in this case) will specify:

< Transfer-Encoding: chunked^M

Where 7.63 will say (with different length values):

< Content-Length: 0^M

The 7.63 version also has this which 7.61 does not show.

* ALPN, offering h2

And in an additional failing connection with 7.63 the following lines are different/new:

< HTTP/1.1 401 Authorization Required^M
< WWW-Authenticate: Negotiate^M
< Vary:Accept-Encoding^M
< Content-Type: text/html; charset=iso-8059-1^M

Previously the content type was application/x-git-upload-pack results. 

I'm happy to test other version or start bisecting if nothing stands out.

**Comment by bagder** [comment]
That would indicate that this problem has something else than `.netrc` parsing involved.

Is this the exact same git code with just different libcurls plugged in?

I would be interested in two things:

1. Does the curl master branch still misbehave
2. If yes, then a bisect would be very valuable

**Comment by justinvreeland** [comment]
I bisected both ways, from good -> bad and from bad -> good. It looks like the problem was introduced with:

```
commit 07ebaf837843124ee670e5b8c218b80b92e06e47                                 
Author: Elia Tufarolo <elia.tufarolo@hcl.com>                                   
Date:   Tue Nov 13 18:30:56 2018 +0100                                          
                                                                                
    http_negotiate: do not close connection until negotiation is completed      
                                                                          
    Fix HTTP POST using CURLAUTH_NEGOTIATE.                                     
                                                                            
    Closes #3275     
```                                                           

and  fixed with:
```
commit ebe658c1e5a6577178981a7f406794699305be5c                                 
Author: Daniel Stenberg <daniel@haxx.se>                                        
Date:   Fri Jan 4 23:34:50 2019 +0100                                           
                                                                                
    Revert "http_negotiate: do not close connection until negotiation is completed"
                                                                                
    This reverts commit 07ebaf837843124ee670e5b8c218b80b92e06e47.               
                                                                                
    This also reopens PR #3275 which brought the change now reverted.           
                                                                                
    Fixes #3384                                                                 
    Closes #3439 
```                                                        

I'll ask the packager to include the fix thanks!
